pixeltable/catalog/catalog.py

```diff
@@ -1724,10 +1816,26 @@ def store_tbl_md(
             assert version_md.tbl_id == str(tbl_id)
             if schema_version_md is not None:
                 assert version_md.schema_version == schema_version_md.schema_version
-            tbl_version_record = schema.TableVersion(
-                tbl_id=tbl_id, version=version_md.version, md=dataclasses.asdict(version_md)
-            )
-            session.add(tbl_version_record)
+            if (
+                session.query(schema.TableVersion)
+                .filter(schema.TableVersion.tbl_id == tbl_id, schema.TableVersion.version == version_md.version)
+                .count()
+                > 0
+            ):
+                # This table version already exists; update it. (This typically happens when setting the is_hidden
+                # flag of an existing TableVersion from True to False.)
```

Comment:
Validate that only the flag gets updated (and in particular that created_at and schema_version are identical).